app/libs/globals.php

```diff
@@ -20,22 +20,27 @@
 if (!defined('APP_JS_PATH')) define('APP_JS_PATH', getenv('APP_JS_PATH'));
 if (!defined('LEAF_DOMAIN')) define('LEAF_DOMAIN', getenv('APP_URL_NEXUS'));
 
-preg_match('(\/.+\/)', $_SERVER['SCRIPT_FILENAME'], $match);
-
-$path = str_replace('/var/www/html', '', $match[0]);
-$path = str_replace('/admin/', '/', $path);
-$path = str_replace('/api/', '/', $path);
-$path = str_replace('/auth_cookie/', '/', $path);
-$path = str_replace('/auth_domain/', '/', $path);
-$path = str_replace('/auth_token/', '/', $path);
-$path = str_replace('/dynicons/', '/', $path);
-$path = str_replace('/login/', '/', $path);
-$path = str_replace('/qrcode/', '/', $path);
-$path = str_replace('/scripts/', '/', $path);
-$path = str_replace('sources/../mailer', '', $path);
-$path = str_replace('/utils/', '/', $path);
-$l_path = trim($path, '/');
-$path = rtrim($path, '/');
-
-if (!defined('PORTAL_PATH')) define('PORTAL_PATH', $path);
-if (!defined('LEAF_NEXUS_URL')) define('LEAF_NEXUS_URL', getenv('APP_URL_NEXUS') . $l_path . '/');
+/*
+    We need to extract the portal url from the SCRIPT_FILENAME so we can get the data from the sites table.
+    There are times where there is another folder tacked on to the end of the url, in those cases that folder
+    needs to be striped from the url
+
+    i.e. /Academy/Demo1/admin
+    I decided it best to put this into a class and have the class deal with it to keep this file clean
+*/
+
+require_once getenv('APP_PATH') . '/Leaf/Db.php';
+require_once getenv('APP_PATH') . '/Leaf/Site.php';
+
+$file_paths_db = new App\Leaf\Db(DIRECTORY_HOST, DIRECTORY_USER, DIRECTORY_PASS, 'national_leaf_launchpad');

```

Comment:
I'm curious if putting a db call in the globals will cause another increase in connections like we faced with the last time we put a db call into one of our config/global. Depends how often this will be called I guess or maybe we can cache it so its not constantly calling it, more looking at the cache.